cockatrice/src/client/ui/picture_loader.cpp

```diff
@@ -43,6 +43,12 @@ PictureToLoad::PictureToLoad(CardInfoPtr _card)
             sortedSets << CardSet::newInstance("", "", "", QDate());
         }
         std::sort(sortedSets.begin(), sortedSets.end(), SetDownloadPriorityComparator());
+        // If the pixmapCacheKey corresponds to a specific set, we have to try to load it first.
+        for (const auto &set : card->getSets()) {
+            if (set.getProperty(QString("uuid")) == card->getPixmapCacheKey()) {
+                sortedSets.prepend(sortedSets.takeAt(sortedSets.indexOf(set.getPtr())));
```

Comment:
There's a lot to this line, maybe try breaking it down a bit to explain what's going on?